python/cugraph-pyg/cugraph_pyg/loader/cugraph_node_loader.py

```diff
@@ -132,26 +162,42 @@ def __init__(
 
         bulk_sampler.flush()
 
-        self.__end_exclusive = 0
-
     def __next__(self):
         # Quit iterating if there are no batches left
-        if self.__next_batch >= self.__num_batches:
+        if self.__next_batch >= self.__num_batches + self.__starting_batch_id:
             raise StopIteration
 
         # Load the next set of sampling results if necessary
         if self.__next_batch >= self.__end_exclusive:
             # Read the next parquet file into memory
-            rank_path = os.path.join(self.__directory.name, f"rank={self.__rank}")
+            dir_path = (
+                self.__directory
+                if isinstance(self.__directory, str)
+                else self.__directory.name
+            )
+            rank_path = os.path.join(dir_path, f"rank={self.__rank}")
```

Comment:
I have been thinking of removing this `rank` behavior for bulk sampler in the next release. I dont think it is adding any value which cant be derived by just changing the dir path.

Reply:
I agree we should remove it in 23.06.  I'll create an issue

Reply:
#3385